src/gadgets/segment_proof.rs

```diff
@@ -0,0 +1,99 @@
+use super::mpt_update::SegmentType;
+use crate::{
+    constraint_builder::{FixedColumn, Query},
+    MPTProofType,
+};
+use halo2_proofs::{
+    arithmetic::FieldExt, circuit::Region, halo2curves::bn256::Fr, plonk::ConstraintSystem,
+};
+
+const SEGMENT_PROOF_DIRECTION_TUPLES: [(SegmentType, MPTProofType, u64); 20] = [
+    // AccountLeaf0
+    (SegmentType::AccountLeaf0, MPTProofType::BalanceChanged, 0),
+    (SegmentType::AccountLeaf0, MPTProofType::CodeHashExists, 0),
+    (SegmentType::AccountLeaf0, MPTProofType::CodeSizeExists, 0),
+    (SegmentType::AccountLeaf0, MPTProofType::NonceChanged, 0),
+    (
+        SegmentType::AccountLeaf0,
+        MPTProofType::PoseidonCodeHashExists,
+        0,
+    ),
+    (SegmentType::AccountLeaf0, MPTProofType::StorageChanged, 0),
+    (
+        SegmentType::AccountLeaf0,
+        MPTProofType::StorageDoesNotExist,
+        0,
+    ),
+    // AccountLeaf1
+    (SegmentType::AccountLeaf1, MPTProofType::BalanceChanged, 0),
+    (SegmentType::AccountLeaf1, MPTProofType::CodeHashExists, 0),
+    (SegmentType::AccountLeaf1, MPTProofType::CodeSizeExists, 0),
+    (SegmentType::AccountLeaf1, MPTProofType::NonceChanged, 0),
+    (SegmentType::AccountLeaf1, MPTProofType::StorageChanged, 0),
+    (
+        SegmentType::AccountLeaf1,
+        MPTProofType::StorageDoesNotExist,
+        0,
+    ),
+    (
+        SegmentType::AccountLeaf1,
+        MPTProofType::PoseidonCodeHashExists,
+        1,
+    ),
+    // AccountLeaf2
+    (SegmentType::AccountLeaf2, MPTProofType::BalanceChanged, 0),
+    (SegmentType::AccountLeaf2, MPTProofType::CodeSizeExists, 0),
+    (SegmentType::AccountLeaf2, MPTProofType::NonceChanged, 0),
+    (SegmentType::AccountLeaf2, MPTProofType::CodeHashExists, 1),
+    (SegmentType::AccountLeaf2, MPTProofType::StorageChanged, 1),
+    (
+        SegmentType::AccountLeaf2,
+        MPTProofType::StorageDoesNotExist,
+        1,
+    ),
+];
```

Comment:
Not sure all of these instructions are correct but it seems a little wired that there is no any directions relating to `AccountLeaf3` and `AccountLeaf4`

Reply:
Add segment-proof-direction tuples for `AccountLeaf3` and `AccountLeaf4` in commit [73d1959](https://github.com/scroll-tech/mpt-circuit/pull/32/commits/73d1959f9b7b46b9651a05565519a361f800ce7a). And a TODO to check and update this constant when implementing the corresponding proof type.